core/src/conn.rs

```diff
@@ -133,6 +138,47 @@ impl MongoConnection {
         }
     }
 
+    fn get_libmongosqltranslate_version() -> Result<String> {
+        let get_mongosqltranslate_version_command = doc! {
+            "command": "getMongosqlTranslateVersion",
+            "options": {},
+        };
+
+        let returned_doc =
+            handle_libmongosqltranslate_command(get_mongosqltranslate_version_command)?;
+
+        let libmongosqltranslate_version = returned_doc
+            .get("version")
+            .expect(
```

Comment:
One general comment, I see quite a few `expect()`s added in the patch.  Take another pass at some of these and see where an adding an error variant would make more sense if it can be returned instead of panicking.